Lib/shutil.py

```diff
@@ -446,43 +466,51 @@ def copytree(src, dst, symlinks=False, ignore=None, copy_function=copy2,
     function that supports the same signature (like copy()) can be used.
 
     """
-    names = os.listdir(src)
+    # TODO: use with statement; omitted just to make code review easier
+    # (will add later)
```

Comment:
It might be worth it to use a subfunction to avoid one indentation level?

with scandir():
  _copytree(...)

Reply:
I like that